### PR TITLE
added typescript insertion_sort algorithm

### PR DESCRIPTION
**Pull request by DanielRomeo**
**Hi!** 
---

**Fixes #{ISSUE_NUMBER}**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes.

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-isrt/typescript/insertion_sort.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const swap = (arr: Array<number>, a: number, b: number)=>{
```
Comment (RavinduSachintha) [inline algorithms/ar-isrt/typescript/insertion_sort.ts:1]: Codacy found an issue: [Missing space before =>.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6355990)

File: algorithms/ar-isrt/typescript/insertion_sort.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        }    
+    } 
+	return values;
+}
+
+
+function main () {
+	let input: number[] = [2, 3, 0, 4];
+	let output: number[] = insertionsort(input);
+	for (let x in output) {
```
Comment (RavinduSachintha) [inline algorithms/ar-isrt/typescript/insertion_sort.ts:27]: Codacy found an issue: [The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6355990)

File: algorithms/ar-isrt/typescript/insertion_sort.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+const swap = (arr: Array<number>, a: number, b: number)=>{
+	let temp = arr[a];
+	arr[a] = arr[b];
+	arr[b] = temp;
+}
+
+const insertionsort = (values: Array<number>)=>{
```
Comment (RavinduSachintha) [inline algorithms/ar-isrt/typescript/insertion_sort.ts:7]: Codacy found an issue: [Missing space before =>.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6355990)

File: algorithms/ar-isrt/typescript/insertion_sort.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+const swap = (arr: Array<number>, a: number, b: number)=>{
+	let temp = arr[a];
+	arr[a] = arr[b];
+	arr[b] = temp;
+}
```
Comment (RavinduSachintha) [inline algorithms/ar-isrt/typescript/insertion_sort.ts:5]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6355990)

File: algorithms/ar-isrt/typescript/insertion_sort.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+const insertionsort = (values: Array<number>)=>{
+
+	for (let i = 1; i < values.length; i++) 
+    { 
+        let j = i; 
+        
+        while (j > 0 && values[j] <  values[j -1]) 
+        { 
+           	 swap(values, j, j - 1);
```
Comment (RavinduSachintha) [inline algorithms/ar-isrt/typescript/insertion_sort.ts:15]: Codacy found an issue: [Mixed spaces and tabs.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6355990)

File: algorithms/ar-isrt/typescript/insertion_sort.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
```
Comment (RavinduSachintha) [inline algorithms/ar-isrt/typescript/insertion_sort.ts:16]: Codacy found an issue: [Mixed spaces and tabs.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6355990)

File: algorithms/ar-isrt/typescript/insertion_sort.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    } 
+	return values;
+}
+
+
+function main () {
+	let input: number[] = [2, 3, 0, 4];
+	let output: number[] = insertionsort(input);
+	for (let x in output) {
+        console.log(output[x]);
```
Comment (RavinduSachintha) [inline algorithms/ar-isrt/typescript/insertion_sort.ts:28]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6355990)

File: algorithms/ar-isrt/typescript/insertion_sort.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+	for (let i = 1; i < values.length; i++) 
+    { 
+        let j = i; 
+        
+        while (j > 0 && values[j] <  values[j -1]) 
+        { 
+           	 swap(values, j, j - 1);
+        	
+        	j--;
```
Comment (RavinduSachintha) [inline algorithms/ar-isrt/typescript/insertion_sort.ts:17]: Codacy found an issue: [Mixed spaces and tabs.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6355990)

File: algorithms/ar-isrt/typescript/insertion_sort.ts [hunk 9]
```diff
@@ -0,0 +1,28 @@
+const swap = (arr: Array<number>, a: number, b: number) =>{
```
Comment (RavinduSachintha) [inline algorithms/ar-isrt/typescript/insertion_sort.ts:1]: Codacy found an issue: [Missing space before =>.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6355990)

File: algorithms/ar-isrt/typescript/insertion_sort.ts [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+	arr[b] = temp;
+};
+const insertionsort = (values: Array<number>) =>{
+	for (let i = 1; i < values.length; i++) 
+    {
+        let j = i; 
+        while (j > 0 && values[j] <  values[j -1]) 
+        {
+            swap(values, j, j - 1);
+        	j--;
```
Comment (RavinduSachintha) [inline algorithms/ar-isrt/typescript/insertion_sort.ts:13]: Codacy found an issue: [Mixed spaces and tabs.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6355990)

File: algorithms/ar-isrt/typescript/insertion_sort.ts [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+        }    
+    }
+	return values;
+}
+function main () {
+	let input: number[] = [2, 3, 0, 4];
+	let output: number[] = insertionsort(input);
+	for (let x in output) {
+        if (output.hasOwnProperty(x)) {
+            console.log(output[x]);
```
Comment (RavinduSachintha) [inline algorithms/ar-isrt/typescript/insertion_sort.ts:23]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6355990)

File: algorithms/ar-isrt/typescript/insertion_sort.ts [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        let j = i; 
+        while (j > 0 && values[j] <  values[j -1]) 
+        {
+            swap(values, j, j - 1);
+        	j--;
+        }    
+    }
+	return values;
+}
```
Comment (RavinduSachintha) [inline algorithms/ar-isrt/typescript/insertion_sort.ts:17]: Codacy found an issue: [Missing semicolon.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6355990)

File: algorithms/ar-isrt/typescript/insertion_sort.ts [hunk 13]
```diff
@@ -0,0 +1,28 @@
+const swap = (arr: Array<number>, a: number, b: number) =>{
+	let temp = arr[a];
+	arr[a] = arr[b];
+	arr[b] = temp;
+};
+const insertionsort = (values: Array<number>) =>{
```
Comment (RavinduSachintha) [inline algorithms/ar-isrt/typescript/insertion_sort.ts:6]: Codacy found an issue: [Missing space before =>.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6355990)

**Comment by Danushka96** [comment]
Hi @DanielRomeo 
Can you update the main Readme.md file with the algorithm you implemented?
Just update the Readme.md file and send the commit to this PR, 
Thanks

**Review by Danushka96 — APPROVED** [review]

**Comment by DanielRomeo** [comment]
@Danushka96  alright

**Comment by DanielRomeo** [comment]
@Danushka96  done!

**Comment by RavinduSachintha** [comment]
@all-contributors please add @DanielRomeo for code and doc

**Comment by RavinduSachintha** [comment]
@DanielRomeo thank you for your contribution. keep it up :+1: make sure to star this repo :wink:

**Pull request merged**